FluidNC/src/Configuration/GenericFactory.h

```diff
@@ -8,19 +8,8 @@
 #include "HandlerBase.h"
 
 namespace Configuration {
-    template <typename BaseType>
-    class GenericFactory {
-        static GenericFactory& instance() {
-            static GenericFactory instance_;
-            return instance_;
-        }
-
-        GenericFactory() = default;
-
-        GenericFactory(const GenericFactory&) = delete;
-        GenericFactory& operator=(const GenericFactory&) = delete;
-
-        class BuilderBase {
+      template <typename BaseType>
+      class BuilderBase {
```

Comment:
Had to make BuilderBase a non-inner class to be able to forward declare it in HandlerBase.h